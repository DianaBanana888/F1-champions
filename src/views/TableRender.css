.table {
  font-family: "Lucida Sans Unicode", "Lucida Grande", Sans-Serif;
  font-size: 14px;
  width: 100%;
  text-align: center;
  border-collapse: collapse;
  background: #252F48;
  margin: 10px;
}
@media (max-width: 699px) {
  .table {
    margin-bottom: 30px;
  }
}
.table th {
  color: #EDB749;
  border-bottom: 1px solid #37B5A5;
  padding: 12px 17px;
}
.table td {
  color: #CAD4D6;
  border-bottom: 1px solid #37B5A5;
  border-right:1px solid #37B5A5;
  padding: 7px 17px;
}
.table tr:last-child td {
  border-bottom: none;
}
.table td:last-child {
  border-right: none;
}
.table tr:hover td {
  text-decoration: underline;
}

.table tr.selected, tr.winner{
  background-color: gold;
}

.table tr.selected td, tr.winner td{
  color: #252F48;
}

@media (min-width: 700px) {
    .winner-seasons-table {
    width: 30%;
  }
}

@media (min-width: 700px) {
    .winner-races-table {
      width: 50%;
    }
}
