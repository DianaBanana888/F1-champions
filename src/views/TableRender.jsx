import React, { useState, useEffect } from 'react';
import WinnersTable from './WinnersTable.jsx'
import ExternalWidget from './ExternalWidget.jsx'
import { getWorldChampionsStandings } from '../ergast/fetchRequest'
import './TableRender.css'
import classNames from 'classnames'

export default function GetRequestHooks() {
  const [worldChampions, setWorldChampions] = useState([]);
  const [activeSeason, setActiveSeason] = useState(null);
  const yearDataStart = 1950;
  const yearStart = 2005;
  const expectedOffset = yearStart - yearDataStart;

  let activeWorldChampion = null;
  activeWorldChampion = worldChampions.find((standings) => standings.season === activeSeason)?.DriverStandings[0]?.Driver?.driverId;

  useEffect(() => {
    getWorldChampionsStandings(expectedOffset).then(data => setWorldChampions(data))
  }, []);

  return (
    <>
      <table className="table winner-seasons-table">
        <WinnersTable worldChampions={worldChampions} activeSeason={activeSeason} setActiveSeason={setActiveSeason} />
      </table>

      <table className={
        classNames(
          activeSeason && 'table',
          "winner-races-table",
        )}>
        <ExternalWidget season={activeSeason} winner={activeWorldChampion} />
      </table>
    </>
  );
}
