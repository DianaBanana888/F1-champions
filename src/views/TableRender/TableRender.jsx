import React, { useState, useEffect } from 'react';
import classNames from 'classnames';
import WinnersTable from '../WinnerTable/WinnersTable';
import ExternalWidget from '../ExternalWidget';
import { getWorldChampionsStandings } from '../../ergast/fetchRequest';
import './TableRender.css';

export default function GetRequestHooks() {
  const [worldChampions, setWorldChampions] = useState([]);
  const [activeSeason, setActiveSeason] = useState(null);
  const yearDataStart = 1950;
  const yearStart = 2005;
  const expectedOffset = yearStart - yearDataStart;

  let activeWorldChampion = null;
  activeWorldChampion = worldChampions
    .find((standings) => standings.season === activeSeason)
    ?.DriverStandings[0]?.Driver?.driverId;

  useEffect(() => {
    getWorldChampionsStandings(expectedOffset).then((data) => setWorldChampions(data));
  }, [expectedOffset]);

  return (
    <>
      <table className="table winner-seasons-table">
        <WinnersTable
          worldChampions={worldChampions}
          activeSeason={activeSeason}
          setActiveSeason={setActiveSeason}
        />
      </table>

      <table className={
        classNames(
          activeSeason && 'table',
          'winner-races-table',
        )
      }
      >
        <ExternalWidget season={activeSeason} winner={activeWorldChampion} />
      </table>
    </>
  );
}
