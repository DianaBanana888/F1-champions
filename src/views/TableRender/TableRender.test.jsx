import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TableRender from './TableRender';

test('render table sceleton for world champions', () => {
  render(<TableRender />);
  const isTable = screen.getAllByRole('table')[0];
  expect(isTable).toBeInTheDocument();
});

it('renders TableRender correctly', () => {
  const tree = renderer
    .create(<TableRender />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
