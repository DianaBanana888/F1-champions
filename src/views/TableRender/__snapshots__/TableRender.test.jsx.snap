// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TableRender correctly 1`] = `
Array [
  <table
    className="table winner-seasons-table"
  >
    <tbody>
      <tr
        className="header"
      >
        <th>
          Season
        </th>
        <th>
          World Champion
        </th>
      </tr>
    </tbody>
  </table>,
  <table
    className="winner-races-table"
  />,
]
`;
